### PR TITLE
UNR-2710 More tests for SpatialWorkerConnection

### PR DESCRIPTION
**Pull request by aleximprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
1. Removed code duplication
2. Added the fixture

#### Tests
Run SpatialWorkerConnection tests

#### Primary reviewers
@improbable-valentyn @mironec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp [hunk 1]
```diff
@@ -199,162 +159,178 @@ bool FFindWorkerResponseOfType::Update()
 	}
 }
 
-DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FCleanupConnection, USpatialWorkerConnection*, Connection);
-bool FCleanupConnection::Update()
+struct FConnectionsFixture;
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp]: Is the forward declaration needed?

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp [hunk 2]
```diff
@@ -199,162 +159,178 @@ bool FFindWorkerResponseOfType::Update()
...
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanupConnectionFixture(ServerConnection, ClientConnection));
+	}
+
+	USpatialWorkerConnection* ClientConnection = nullptr;
+	USpatialWorkerConnection* ServerConnection = nullptr;
+};
+
+bool FSetupConnectionFixture::Update()
+{
+	ClientConnection->AddToRoot();
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp]: Does this need to be in a latent command? I think we could do it directly in the ConnectionFixture constructor. I think there is a very small chance the ServerConnection and ClientConnection could be deallocated in the meantime.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp (hunk 1 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp]: Good point, not needed anymore.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp (hunk 2 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp]: It should be fine to have it in constructor instead 👍

Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp]: Addressed.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp (hunk 1 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp]: Addressed.

**Pull request closed without merging**